tests/tools/TestApp/src/mcm.c

```diff
@@ -1,104 +1,104 @@
-    int err = mesh_create_client(&(mcm->client), cfg);
-    if (err) {
-        printf("Failed to create mesh client: %s (%d)\n", mesh_err2str(err), err);
-        exit(1);
-    }
-    return err;
-}
-
-int mcm_create_tx_connection(mcm_ts* mcm, const char* cfg){
-    int err = mesh_create_tx_connection(mcm->client, &(mcm->connection), cfg);
-    if (err) {
-        printf("Failed to create connection: %s (%d)\n", mesh_err2str(err), err);
-        mesh_delete_client(&(mcm->connection));
-    }
-    return err;
-}
-
-
-int mcm_create_rx_connection(mcm_ts* mcm, const char* cfg){
-    int err = mesh_create_rx_connection(mcm->client, &(mcm->connection), cfg);
-    if (err) {
-        printf("Failed to create connection: %s (%d)\n", mesh_err2str(err), err);
-         mesh_delete_client(&(mcm->connection));
-    }
-    return err;
-}
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "mcm.h"
+#include "mesh_dp.h"
 
+/* PRIVATE */
+void buffer_to_file(FILE *file, MeshBuffer *buf);
 
-int mcm_send_video_frame(mcm_ts* mcm, const char* frame, int frame_len ){
+int mcm_send_video_frames(MeshConnection *connection, MeshClient *client, FILE *file) {
     int err = 0;
     MeshBuffer *buf;
-
-    /* Ask the mesh to allocate a shared memory buffer for user data */
-    err = mesh_get_buffer(mcm->connection, &buf);
-    if (err) {
-        printf("Failed to get buffer: %s (%d)\n", mesh_err2str(err), err);
+    if (file == NULL) {
+        printf("[TX] Failed to serialize video: file is null \n");
+        exit(1);
     }
 
-    /* Fill the buffer with user data */
-    buf->payload_ptr = frame;
-    put_user_video_frames(buf->payload_ptr, frame_len);
-
-    /* Send the buffer */
-    err = mesh_put_buffer(&buf);
-    if (err) {
-        printf("Failed to put buffer: %s (%d)\n", mesh_err2str(err), err);
-    }
-    err = mesh_shutdown_connection(&(mcm->connection));
-    if (err){
-        printf("Failed to shutdown connection: %s (%d)\n", mesh_err2str(err), err);
+    unsigned char frame_num = 0;
+    size_t read_size = 1;
+    while (1) {
+
+        /* Ask the mesh to allocate a shared memory buffer for user data */
+        err = mesh_get_buffer(connection, &buf);
+        if (err) {
+            printf("[TX] Failed to get buffer: %s (%d)\n", mesh_err2str(err), err);
+            exit(err);
+        }
+        read_size = fread(buf->payload_ptr, 1, buf->payload_len, file);
+        if (read_size == 0) {
+            exit(2);
+        }
+
+        /* Send the buffer */
+        printf("[TX] Sending frame: %d\n", ++frame_num);
+        err = mesh_put_buffer(&buf);
+        if (err) {
+            printf("[TX] Failed to put buffer: %s (%d)\n", mesh_err2str(err), err);
+            exit(err);
+        }
+
+        /* Temporary implementation for pacing */
+        /* TODO: Implement pacing calculation */
+        usleep(40000);
     }
+    printf("[TX] data sent successfully \n");
     return err;
```

Comment:
If succeeded, maybe it is worth returning 0?
```suggestion
    printf("[TX] data sent successfully \n");
    return 0;
```

Reply:
done